File: file_io/file_io.rb
require 'tempfile'

data = "x" * 1_000_000

tmpfile = Tempfile.new('bench')
path = tmpfile.path
tmpfile.close

start = Process.clock_gettime(Process::CLOCK_MONOTONIC)

100.times do
  File.write(path, data)
  File.read(path)
end

elapsed = Process.clock_gettime(Process::CLOCK_MONOTONIC) - start

File.delete(path) if File.exist?(path)

puts "Time: #{'%.3f' % elapsed}s"

File: file_io/file_io.py
import os
import tempfile
import time

data = "x" * 1_000_000

fd, path = tempfile.mkstemp()
os.close(fd)

start = time.monotonic()

for _ in range(100):
    with open(path, "w") as f:
        f.write(data)
    with open(path, "r") as f:
        f.read()

elapsed = time.monotonic() - start

os.remove(path)

print(f"Time: {elapsed:.3f}s")
